openmetrics/src/main/java/io/airlift/openmetrics/MetricsResource.java

```diff
@@ -145,6 +147,8 @@ private String mBeanNameToMetricName(ObjectName objectName, String attributeName
         StringBuilder metricNameBuilder = new StringBuilder("JMX_")
                 .append(objectName.getDomain().replace(".", "_"));
 
+        // TODO name and type are not necessary here since they're included in labels
```

Comment:
Need to solve this. Also, I did not add any tests for non-Airlift MBeans.

Reply:
I'm not sure what's the purpose of the filter. It allows fetching any JMX metric, even those not present in `openmetrics.jmx-object-names`. Parsing the names is complicated and has no test coverage. I couldn't find any referenced about this in the OpenMetrics spec or Prometheus handler.

Reply:
It's exactly for grabbing metrics from jmx without having to specify the objectname config, it's basically the most efficient route for us to capture the jvm garbage collection metrics and such for trino, if we allow all the jmx metrics from the jvm we'll be scraping a very large amount of metrics and only using a very small subset.